src/main/webapp/components/MultiTab.js

```diff
@@ -1,26 +1,39 @@
 import React, { Component } from 'react';
 import { hh, h } from 'react-hyperscript-helpers';
 import { Tabs, Tab } from 'react-bootstrap';
-
+import { isEmpty } from "../util/Utils";
 
 export const MultiTab = hh(class MultiTab extends Component {
 
   constructor(props) {
     super(props);
 
     this.state = {
-      key: ""
+      key: "",
+      defaultActive: ""
     };
 
     this.handleSelect = this.handleSelect.bind(this);
   }
 
- handleSelect(key) {
+  handleSelect(key) {
     this.setState({ key: key });
+    if (this.props.handleSelect !== undefined) {
+      this.props.handleSelect(key);
+    }
   }
 
   componentDidMount() {
-    this.setState({key: this.props.defaultActive});
+    this.setState({key: this.props.defaultActive, defaultActive: this.props.defaultActive});
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
This class seems to be a good candidate to be refactored as a functional component, specially to get rid of getDerivedStateFromProps. Not required now but may be a good exercise.

Reply:
This change might modify how it is used in other components, and would require re-test the branch. I can implement this in the branch i'm currently working on. Let me know if this sounds good for you.